projects/saturn/src/main/java/io/fairspace/saturn/services/views/ViewStoreClientFactory.java

```diff
@@ -103,62 +102,103 @@ void ensureTableExists(Table table) throws SQLException {
-                        .map(column -> column.name)
-                        .collect(Collectors.joining(", "));
-                var command = String.format("create table %s ( %s, primary key ( %s ) )", table.name, columnSpecification, keys);
-                log.debug(command);
-                connection.createStatement().execute(command);
-                connection.setAutoCommit(false);
-                log.info("Table {} created.", table.name);
+                createTable(table, connection);
             } else {
-                var columnMetadata = getColumnMetadata(connection, table.name);
-                var newColumns = table.columns.stream()
-                        .filter(column -> !column.type.isSet() && !columnMetadata.containsKey(column.name))
-                        .collect(Collectors.toList());
-                try {
-                    log.debug("New columns: {}", new ObjectMapper().writeValueAsString(newColumns));
-                } catch (JsonProcessingException e) {
-                    e.printStackTrace();
-                }
-                // Update existing table
-                if (!newColumns.isEmpty()) {
-                    connection.setAutoCommit(true);
-                    var command = newColumns.stream()
-                            .map(column -> String.format("alter table %s add column %s %s",
-                                    table.name, column.name, databaseTypeForColumnType(column.type)))
-                            .collect(Collectors.joining("; "));
-                    log.debug(command);
-                    connection.createStatement().execute(command);
-                    connection.setAutoCommit(false);
-                    log.info("Table {} updated.", table.name);
-                }
+                updateTable(table, connection);
             }
         }
     }
 
+    void ensureJoinTableExists(Table table) throws SQLException {
+        try (var connection = getConnection()) {
+            log.debug("Check if table {} exists ...", table.name);
+            var resultSet = connection.getMetaData().getTables(null, null, table.name, null);
+            var tableExists = resultSet.next();
+            
+            if (!tableExists) {
+                createTable(table, connection);
+                createIndexes(table, connection);
+            } else {
+                updateTable(table, connection);
+            }
+        }
+    }
+
+    private void updateTable(Table table, Connection connection) throws SQLException {
+        var columnMetadata = getColumnMetadata(connection, table.name);
+        var newColumns = table.columns.stream()
+                .filter(column -> !column.type.isSet() && !columnMetadata.containsKey(column.name))
+                .collect(Collectors.toList());
+        try {
+            log.debug("New columns: {}", new ObjectMapper().writeValueAsString(newColumns));
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
```

Comment:
could you please replace it using log.error? I saw it many times, but once you are touching this part

Reply:
"I saw it many times", you mean in general or Fairspace code? I think it's rare but I will change!

Reply:
its an exception if a 'log.debug' doesn't work out, that should not be reported as an error

Reply:
as discussed, let's log an error and rethrow, because it should never fail indeed